AraEvent/AraEventCalibrator.cxx

```diff
@@ -896,226 +1030,373 @@ void AraEventCalibrator::calibrateEvent(UsefulAtriStationEvent *theEvent, AraCal
-            for(Int_t chan=0;chan<5;chan++) {
-                Int_t chanId=chan+RFCHAN_PER_DDA*dda;
-                voltMapIt=theEvent->fVolts.find(chanId);
-                Int_t chanId2=5+RFCHAN_PER_DDA*dda;
-                voltMapIt2=theEvent->fVolts.find(chanId2);
-    
+}
+
+//! Voltage calibration
+/*!
+    \param theEvent the useful atri event pointer
+    \param voltMapIt the iterator referring to the voltage elements in the 2d map container
+    \param stationId id of the station
+    \return void
+*/
+void AraEventCalibrator::VoltageCalibration(UsefulAtriStationEvent *theEvent, std::map< Int_t, std::vector <Double_t> >::iterator &voltMapIt, std::vector<std::vector<int> > *sampleList, AraStationId_t thisStationId)
+{
+    int blockIndex = 0;///< This is now filled in the code below- KAH
+    int sampleIndex, sampleNumber = 0; ///< removed this and instead use "samp" variable below. re-use this for the trimmed sample index -MK-
+    int samples_per_block = SAMPLES_PER_BLOCK;
+
+    for(int dda=0;dda<DDA_PER_ATRI;dda++) {
+        for(Int_t chan=0;chan<RFCHAN_PER_DDA;chan++) {
+            Int_t chanId=chan+RFCHAN_PER_DDA*dda;
+            voltMapIt=theEvent->fVolts.find(chanId);
+            if(voltMapIt!=theEvent->fVolts.end()) {
                 Int_t numPoints=(voltMapIt->second).size();
                 for(int samp=0;samp<numPoints;samp++) {
-                    voltMapIt->second[samp]-=voltMapIt2->second[samp];
+                    sampleIndex = sampleList->at(chanId)[samp]; ///< capacitor sample index
+                    sampleNumber = sampleIndex%samples_per_block; ///< block sample index
+                    blockIndex = int(sampleIndex/samples_per_block); ///< capacitor block index
+                    //! Move to conversion function
+                    //! Apply conversion parameter on each sample
+                    voltMapIt->second[samp] = convertADCtoMilliVolts( voltMapIt->second[samp], dda, blockIndex, chan, sampleNumber, thisStationId);
                 }
             }
         }
     }
+}
+
+//! Inverts only RF channels = 0,4,8 in A3
+//! Apply Brian's conditioner inside of calibration
+/*!
+    \param theEvent the useful atri event pointer
+    \param voltMapIt the iterator referring to the voltage elements in the 2d map container
+    \param stationId id of the station
+    \return void
+*/
+void AraEventCalibrator::InvertA3Chans(UsefulAtriStationEvent *theEvent, std::map< Int_t, std::vector <Double_t> >::iterator &voltMapIt, AraStationId_t thisStationId)
+{
     
-    // RJN change 13-Feb-2013
-    // Now do zero mean,   
-    
-    //std::cout << "here's a test " << hasZeroMean(calType) << ", " << hasVoltCal(calType) << std::endl;
-    // THM added 12-Feb-2014
-    // After zeroMean (and only then!!) do voltage calibration,
-
-    //KAH added 09-15-2020
-    //first, need to do voltage calibration. if you zero-mean the block beforehand, the ADC values get screwed up and the resulting waveform looks bad.
```

Comment:
What happened to these comments from kaeli? I can't tell in the diff viewer if they're still there.

Reply:
Oh! That's my mistake let me put it back in